infra/gcp/terraform/backend.tf

```diff
@@ -3,6 +3,14 @@ resource "google_service_account" "backend" {
   display_name = "backend"
 }
 
+resource "google_project_iam_binding" "backend_iam" {
+  depends_on = [google_service_account.backend]
+  role    = "roles/cloudsql.client"
+  members = [
+    "serviceAccount:${google_service_account.backend.email}"
+  ]
+}
+
```

Comment:
ah nice! to make it work I also had to give permissions to read from the secret storage, are you sure this works without that permission?

Reply:
It works without that extra permission. The difference I think is in the new account you needed to create, while this is adjusting existing accounts with the TF config. Probably the CloudRun service accounts already have the requires roles to read from secret store.

Reply:
Ah could be, I found find it weird though because I'm not sure why it would have implicitly one permission but not the other, but I guess this could very well be the case. Just to confirm, the secret would only be used when actually connecting to the DB, possibly just from the backend (and not db-manager). So, you're saying that this works without this permission when connecting to the DB from the backend. Do we have an API which hits the DB to confirm that it works? Because the fact we didn't have one in Titus by default was the reason why we didn't notice that the db connection wouldn't work.

Reply:
There is an `/healthcheck` route in the backend that responds with a json. This route actually makes a call to the database to check connectivity. The resulting `db` property can be either `ok` or `fail`.
For the `db-manager`, you can POST `/db/migrate` which will either return 200 or else fail.

Both services are configured fine now after added TF config.